File: app/android/app/src/main/java/com/magicvector/domain/model/agent/AgentChatModel.java
package com.magicvector.domain.model.agent;


import com.data.domain.Do.ChatMessageEntity;

import java.util.ArrayList;
import java.util.List;

/**
 * @author 13225
 * @date 2025/10/15 11:25
 */
public class AgentChatModel {
    public AgentModel agentModel;
    public Long lastChatTime = 0L;
    // 最新的10条消息
    public List<ChatMessageEntity> lastChatMessages = new ArrayList<>();
    // 未读消息
    public Integer unreadCount = 0;
}

File: app/android/app/src/main/java/com/magicvector/domain/model/agent/AgentChatModel.kt
package com.magicvector.domain.model.agent

import com.magicvector.domain.vo.agent.AgentChatVO
import com.magicvector.utils.sort.SortItem

/**
 * Agent Item 业务模型
 * @see com.magicvector.domain.entity.AgentChatEntity
 */
class AgentChatModel : SortItem {
    var agentChatVo: AgentChatVO? = null
    var agentId: Long = 0L
    var userId: Long = 0L
    var lastChatTime: Long = 0L
    var updatedAt: Long = 0L

    override fun getUid(): Long {
        return agentId
    }

    override fun getTimestamp(): Long {
        return lastChatTime
    }

    override fun getStringIndex(): String {
        return agentChatVo?.agentVo?.name ?: throw NullPointerException("agentVo.name must not be null")
    }
}